### ISSUE TITLE
Do not remove ROLE_PARTICIPANT but make it inactive

### ISSUE THREAD

**Issue by blootsvoets** [RADAR-base/ManagementPortal#218]
Instead of removing the `ROLE_PARTICIPANT` role whenever a user is reassigned, it should be set to inactive. That way the REST API has a way of knowing whether a user was ever assigned to a given project. The implementation could have two parts:
1. the JWT only contains roles that are active, so participants will no longer be able to submit data for an inactive project.
2. the API exposes inactive participants, something like:
`GET /projects/radartest/participants`
```json
{
   "participants": [...],
   "inactiveParticipants": [...]
}
```

**Comment by nivemaham** [comment]
So after a discussion, we thought having a separate role with the `ROLE_INACTIVE_PARTICIPANT` would be an easy and feasible solution to get this done and we will implement revisions for further history auditing. When a subject is moved from one project to another the subject will get an additional role with the old project and `ROLE_INACTIVE_PARTICIPANT` authority. This authority will limit participant from being able to send data, but can still allow to read data. 
As an SYS_ADMIN , a user can move subject from one project to another and can view inactive projects of subject if available. The default project view would list only active patients. 
As suggested, the JWT will include only active role, which means avoid `INACTIVE_PARTICIPANT` role if available.
The API is formed with additional query parameter `withInactiveParticipants=true`, if not explicitly requested with the query parameter, default response will only contain active subjects and all subjects will be in the same list. This will allow interested clients to get inactive participants with minimal changes to existing implementation.
Any improvements or comments? @dennyverbeeck @yatharthranjan

**Comment by yatharthranjan** [comment]
Sounds like a good approach. Will the subject DTO also contain information about previous projects that the subject was assigned to ?

**Comment by nivemaham** [comment]
Yes. it will contain `roles` which will include `projectName` and `authority`

**Comment by nivemaham** [comment]
After an off-line discussion with  @dennyverbeeck , We decided to add the inactive_participant role to the JWT token as well, however, there is no permissions associated to this role.

**Comment by blootsvoets** [comment]
This was fixed in #243
